Source/RDTechFilter/RDTechFilter.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+
+namespace RealismOverhaul
+{
+    public class RDTechFilters
+    {
+        public class Filter
+        {
+            public string filterID;
+            public Func<AvailablePart, bool> criteria;
+
+            public Filter(string filterID, Func<AvailablePart, bool> criteria)
+            {
+                this.filterID = filterID;
+                this.criteria = criteria;
+            }
+        }
+
+        public class FilterList
```

Comment:
Not a fan of making a custom collection for storing filters. Better to use a collection builtin to the language and if you want those Add, remove etc methods then implement them on `RDTechFilters` instead.